### PR TITLE
Fix incorrect gas limit computation on Arbitrum

### PR DESCRIPTION
**Pull request by onyb**
## Overview

Please see the linked issue for details on the proposed fix. Also added a commit to bubble up errors raised inside `ApproveTransaction()` to the frontend - this was necessary for investigation and the demo.

Resolves https://github.com/brave/brave-browser/issues/21084.

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### Reproducing the bug

ℹ️ Checkout commit 6d3596e5fcbb675e26eaf25953d29ab62e4f7c24.

https://user-images.githubusercontent.com/3684187/154093152-38a1cb37-87c4-4eab-a937-280e68c8eefe.mov

### After the fix

ℹ️ Checkout the branch `f/wallet/arbitrum-bug`.

https://user-images.githubusercontent.com/3684187/154091647-694cf386-9cd2-4ca0-9389-57f7ce4cf26d.mov

### CODE REVIEW

**Comment by bbondy** [comment]
looks like some browser tests need to be updated:
`brave/browser/brave_wallet/send_transaction_browsertest.cc` possibly others too.

**Review by Douglashdaniel — APPROVED** [review]
++ for Front-end

File: browser/brave_wallet/brave_wallet_provider_impl_unittest.cc [hunk 1]
```diff
@@ -743,9 +749,13 @@ TEST_F(BraveWalletProviderImplUnitTest, AddAndApproveTransaction) {
...
   SetInterceptor("{\"jsonrpc\":\"2.0\",\"id\":1,\"result\":\"0x0\"}");
 
-  EXPECT_TRUE(ApproveTransaction(infos[0]->id));
+  mojom::ProviderError error = mojom::ProviderError::kUnknown;
+  std::string error_message;
+
+  EXPECT_TRUE(ApproveTransaction(infos[0]->id, &error, &error_message));
   base::RunLoop().RunUntilIdle();
 
+  EXPECT_EQ(error, mojom::ProviderError::kSuccess);
```
Comment (yrliou) [inline browser/brave_wallet/brave_wallet_provider_impl_unittest.cc:758]: add EXPECT_TRUE(error_message.empty());

File: components/brave_wallet/browser/eth_tx_manager_unittest.cc [hunk 2]
```diff
@@ -942,8 +945,11 @@ TEST_F(EthTxManagerUnitTest, ProcessHardwareSignatureFail) {
   callback_called = false;
   eth_tx_manager()->ProcessHardwareSignature(
       tx_meta_id, "0x00", "9ff044f89c205dd76a194f8b11f50d2eade744e", "",
-      base::BindLambdaForTesting([&](bool success) {
+      base::BindLambdaForTesting([&](bool success,
+                                     mojom::ProviderError error_out,
+                                     const std::string& error_message_out) {
         EXPECT_FALSE(success);
+        EXPECT_EQ(error_out, mojom::ProviderError::kInternalError);
```
Comment (yrliou) [inline components/brave_wallet/browser/eth_tx_manager_unittest.cc:952]: check if error message match the specific msg, or at least make sure it's not empty.

File: components/brave_wallet/browser/eth_tx_manager_unittest.cc [hunk 3]
```diff
@@ -956,8 +962,11 @@ TEST_F(EthTxManagerUnitTest, ProcessHardwareSignatureFail) {
   callback_called = false;
   eth_tx_manager()->ProcessHardwareSignature(
       "-1", "0x00", "9ff044f89c205dd76a194f8b11f50d2eade744e", "",
-      base::BindLambdaForTesting([&](bool success) {
+      base::BindLambdaForTesting([&](bool success,
+                                     mojom::ProviderError error_out,
+                                     const std::string& error_message_out) {
         EXPECT_FALSE(success);
+        EXPECT_EQ(error_out, mojom::ProviderError::kResourceNotFound);
```
Comment (yrliou) [inline components/brave_wallet/browser/eth_tx_manager_unittest.cc:969]: DITTO

File: android/java/org/chromium/chrome/browser/crypto_wallet/fragments/ApproveTxBottomSheetDialogFragment.java [hunk 4]
```diff
@@ -352,8 +353,10 @@ private void approveTransaction() {
         if (txService == null) {
             return;
         }
-        txService.approveTransaction(CoinType.ETH, mTxInfo.id, success -> {
+        txService.approveTransaction(CoinType.ETH, mTxInfo.id, (success, error, errorMessage) -> {
             assert success;
+            Utils.warnWhenError(ApproveTxBottomSheetDialogFragment.TAG_FRAGMENT,
+                    "approveTransaction", error, errorMessage);
```
Comment (SergeyZhukovsky) [inline android/java/org/chromium/chrome/browser/crypto_wallet/fragments/ApproveTxBottomSheetDialogFragment.java:359]: As I understood we need to show that error in UI right? `warnWhenError` just logs it. You can create a follow up to show that on UI on Android

**Review by SergeyZhukovsky — APPROVED** [review]
Android part looks good!

**Review by yrliou — APPROVED** [review]
LGTM

**Comment by srirambv** [comment]
Verification passed on 

Brave | 1.37.58 Chromium: 99.0.4844.35 (Official Build) nightly (64-bit)
-- | --
Revision | `f60a827ddb87f1c403e07713751a5551d5856ac0-refs/branch-heads/4844@{#579}`
OS | Windows 11 Version Dev (Build 22543.1000)

- Verified steps from PR description

https://user-images.githubusercontent.com/17010094/154624526-9c0a1206-fa85-4699-a9fe-214a0283ef9b.mov

**Pull request merged**